javascript/009-amusement-park/amusement-park.js

```diff
@@ -0,0 +1,62 @@
+/// <reference path="./global.d.ts" />
+// @ts-check
+
+/**
+ * Creates a new visitor.
+ *
+ * @param {string} name
+ * @param {number} age
+ * @param {string} ticketId
+ * @returns {Visitor} the visitor that was created
+ */
+export function createVisitor(name, age, ticketId) {
+  return { name, age, ticketId };
+}
+
+/**
+ * Revokes a ticket for a visitor.
+ *
+ * @param {Visitor} visitor the visitor with an active ticket
+ * @returns {Visitor} the visitor without a ticket
+ */
+export function revokeTicket(visitor) {
+  visitor.ticketId = null;
+  return visitor;
+}
+
+/**
+ * Determines the status a ticket has in the ticket tracking object.
+ *
+ * @param {Record<string, string|null>} tickets
+ * @param {string} ticketId
+ * @returns {string} ticket status
+ */
+export function ticketStatus(tickets, ticketId) {
+  if (tickets[ticketId] === undefined) {
+    return 'unknown ticket id';
+  }
+
+  return tickets[ticketId] !== null ? `sold to ${tickets[ticketId]}` : 'not sold';
```

Comment:
Don't mix if-statements and if-expressions together, that looks weird and is hard to read.